UDX-218, UDX-207: Document rest of Cypress App (#4554)
cypress-io/cypress-documentation: content/guides/core-concepts/cypress-app.md

## <Icon name="graduation-cap"></Icon> What you'll learn

- The names and purposes of the visual parts of the Cypress app
- How to use the Selector Playground for targeting your page elements
- How to debug tests using the built-in features of the Cypress app

</Alert>

## Overview

The Cypress desktop app has powerful features for creating, configuring,
browsing, and most importantly running your tests. We'll touch on a few of the
different functions before moving on to the Test Runner, the heart of the
Cypress testing experience.

<Alert type="info">

<strong class="alert-header">Dashboard Integration</strong>

> A number of the features described below will only be shown if you are logged
> into the [Cypress Dashboard](/guides/dashboard/introduction). If you're not sure
> whether you're logged in, you can always check using the profile control in the
> top right of the window.

Reviewer: ```suggestion
When you configure Cypress to record tests to the [Dashboard](/guides/dashboard/introduction), you will get data from your latest recorded runs in the Cypress app to increase visibility into your test history. If you're not sure whether you're logged in, you can always check using the profile control in the
top right of the window.
```

Reviewer: @JessamynSweet @debrisapron let me know what you think of this wording - still playing with it

Author: I think this is pretty good, gonna add this for now

> As the data used by these features are pulled from our cloud servers, they will

Reviewer: ```suggestion
As the data used by these features is pulled from the Cypress Dashboard, they will
```

Once you get comfortable with Cypress you might find you don't need the
Launchpad any more. In this case
[you can run `cypress open` with the `--browser` and `--component` or `--e2e` options](/guides/guides/command-line#cypress-open)
to go straight to the Spec Explorer.

### The Spec Explorer

TODO Image with rest of columns
<DocsImage src="/img/guides/core-concepts/cypress-app/spec-explorer.png" alt="The Spec Explorer"></DocsImage>

On choosing your browser in the Launchpad, you'll be presented with a list of
your specs, their names, locations, and associated metadata columns. Here you
can launch specs by clicking them, create new blank or example specs, or search
specs by name (handy for large test suites).

There are currently three spec metadata columns.

#### Last updated

This column shows when the code of this spec was last updated, using your local
system's own file metadata. This metadata will vary from machine to machine, as
it reflects the state of the code as stored on that computer.

#### Latest runs

This column shows the last time the spec was run, and what status was returned.
This metadata comes from the Dashboard (see [note above](#Overview)) so should

> be the same everywhere.

Reviewer 2: This language suggest that only the very last run is shown, when we actually show the status of the last 4 runs for the current branch and more details about the last run, which is displayed in a tooltip and both the status dots and the tooltip link to the run results for that spec

Author: OK changing to "and the last four returned statuses."

#### Average duration

This column shows the average time taken to run the entire spec file. Again,

> this metadata comes from the Dashboard.

Reviewer 2: It may be helpful to explain that the average is over the last 4 runs for the current branch. Also, in case the spec had multiple groups, the average will reflect the average across runs for the average of durations for groups in each run.

## <Icon name="graduation-cap"></Icon> What you'll learn

- The names and purposes of the visual parts of the Cypress app
- How to use the Selector Playground for targeting your page elements
- How to debug tests using the built-in features of the Cypress app

</Alert>

## Overview

The Cypress desktop app has powerful features for creating, configuring,
browsing, and most importantly running your tests. We'll touch on a few of the
different functions before moving on to the Test Runner, the heart of the
Cypress testing experience.

<Alert type="info">

<strong class="alert-header">Dashboard Integration</strong>

When you configure Cypress to record tests to the
[Dashboard](/guides/dashboard/introduction) you'll see data from your latest

> recorded runs in the Cypress app, increasing visibility into your test history.

Reviewer: ```suggestion
recorded runs in the Cypress app. This increased visibility into your test history allows
you to debug your tests faster and more effectively, all within your local workflow.
```

Reviewer: Also maybe add a line break after this sentence.

The Launchpad is your portal to Cypress, helping with onboarding, choosing a
testing type and launching a browser.

Once you get comfortable with Cypress you might find you don't need the
Launchpad any more. In this case
[you can run `cypress open` with the `--browser` and `--component` or `--e2e` options](/guides/guides/command-line#cypress-open)
to go straight to the Spec Explorer.

### The Spec Explorer

TODO Image with rest of columns
<DocsImage src="/img/guides/core-concepts/cypress-app/spec-explorer.png" alt="The Spec Explorer"></DocsImage>

On choosing your browser in the Launchpad, you'll be presented with a list of
your specs, their names, locations, and data from your latest recorded runs.
Here you can launch specs by clicking them, create new blank or example specs,
or search specs by name (handy for large test suites).

We've made the following spec metadata available directly in the Spec Explorer:

#### Last updated

This column shows when the code of this spec was last updated, using your local
system's own metadata. This metadata will vary from machine to machine, as it

> reflects the state of the code as stored on that computer.

Reviewer: This should definitely call out that 'Last updated' uses git if available in your project, and falls back to system data otherwise.

#### Latest runs

This column shows the last time the spec was run, and the last four returned
statuses. This metadata comes from the Dashboard (see [note above](#Overview))
so should be the same everywhere.

#### Average duration

This column shows the average time taken to run the entire spec file, over the

> last four runs. Again, this metadata comes from the Dashboard.

Reviewer: These sections should focus more on the value that the data provides.

Reviewer: Value narrative notes:
- **Last updated** showing the change history of your tests so you can quickly find the most relevant specs as you’re browsing your test suite
- **Latest runs** so you can monitor, run, and fix tests locally within CI workflows, and then further dig in to test results by clicking through to the Cypress Dashboard
- **Average duration** of your specs so that you can quickly identify and take action to improve the performance of your long-running specs

### Project Runs

TODO Ideally runs would show different commits
<DocsImage src="/img/guides/core-concepts/cypress-app/recorded-runs.png" alt="Recorded Runs"></DocsImage>

This screen shows detailed information about the most recently recorded
[test runs](/guides/dashboard/runs) across all git branches, latest first. This

> metadata comes from the Dashboard (see [note above](#Overview)) so should be the

Reviewer: I would not refer to this as metadata. In fact, I would try to use the word "metadata" as little as possible in this page, since it minimizes the value of the Dashboard.

Reviewer: information, data, insights

Reviewer: "This run overview data"

## Overview

The Cypress App is a desktop application with powerful features for creating,
configuring, browsing and most importantly running your tests. We'll touch on a
few of the different functions before moving on to the Test Runner, the heart of
the Cypress testing experience.

<Alert type="warning">

<strong class="alert-header">Dashboard Syndication</strong>

> A number of the features described below will only be shown if you are logged

Reviewer 3: I wouldn't want to be too sales-y, but perhaps we can add a descriptive statement here for why it's valuable for users to have this data here, e.g.,

```suggestion
As the Cypress app runs tests, it records result data to the Cypress Dashboard. Dashboard derives insights to help you measure and optimize the entire development process. 
```

Author: I've added something in my latest changes and put this on the to-do list for resolving.

top right of the window.

As the data used by these features are syndicated from our cloud servers, they
will only reflect test results that were
[recorded](/guides/dashboard/projects#Set-up-a-project-to-record).

</Alert>

### The Launchpad

<DocsImage src="/img/guides/core-concepts/cypress-app/the-launchpad.png" alt="The Launchpad"></DocsImage>

The Launchpad is your portal to Cypress, helping with onboarding, choosing a
testing type and launching a browser.

Once you get comfortable with Cypress you might find you don't need the
Launchpad any more. In this case you can launch the App with
[`cypress open --e2e` or ` --component`](/guides/guides/command-line#cypress-open)
to go straight to the Spec Explorer.

### The Spec Explorer

TODO Image with rest of columns
<DocsImage src="/img/guides/core-concepts/cypress-app/spec-explorer.png" alt="The Spec Explorer"></DocsImage>

On choosing your browser in the Launchpad, you'll be presented with a list of
your specs, their names, locations, and associated metadata columns. Here you
can launch specs by clicking them, create new blank or example specs, or search
specs by name (handy for large test suites).

> There are currently three spec metadata columns.

Reviewer: How about: "We've made the following data available directly in the Spec Explorer:"

> There are currently three spec metadata columns.

Author: Going with "We've made the following spec metadata available directly in the Spec Explorer"

> #### Last updated

Author: Need to think about this a bit more so have added it to the to-do

This column shows when the code of this spec was last updated, using your local
system's own file metadata. This metadata will vary from machine to machine, as

> it reflects the state of the code as stored on that computer.

Author: OK I'm gonna remove the word file :D

> it reflects the state of the code as stored on that computer.

Author: I rewrote it a little more
